src/compiler/types.rs

```diff
@@ -890,22 +941,40 @@ impl PropertyAssignment {
     }
 }
 
+pub trait HasTypeParametersInterface {
```

Comment:
I don't think this corresponds to an `interface` in the Typescript version but it looks like there are a bunch of AST node types that have an optional `typeParameters?: NodeArray<TypeParameterDeclaration>` field so probably will make sense to have them all implement this shared interface for reading that field